File: Learning Charts/iOSChartsDemo in Objective-C/iOSChartsDemo in Objective-C/ViewController.m
//
//  ViewController.m
//  iOSChartsDemo in Objective-C
//
//  Created by Wes Van der Klomp on 5/3/16.
//  Copyright © 2016 wvdk. All rights reserved.
//

#import "ViewController.h"
@import Charts;

@interface ViewController ()

@property (weak, nonatomic) IBOutlet BarChartView *barChartView;
@property NSArray *months;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    
    _months = [NSArray arrayWithObjects: @"Jan", @"Feb", @"Mar", @"Apr", @"May", @"Jun", @"Jul", @"Aug", @"Sep", @"Oct", @"Nov", @"Dec", nil];
    
    NSArray *unitsSold = [NSArray arrayWithObjects:
                          [NSNumber numberWithFloat:20.0],
                          [NSNumber numberWithFloat:4.0],
                          [NSNumber numberWithFloat:6.0],
                          [NSNumber numberWithFloat:3.0],
                          [NSNumber numberWithFloat:12.0],
                          [NSNumber numberWithFloat:16.0],
                          [NSNumber numberWithFloat:4.0],
                          [NSNumber numberWithFloat:18.0],
                          [NSNumber numberWithFloat:2.0],
                          [NSNumber numberWithFloat:4.0],
                          [NSNumber numberWithFloat:5.0],
                          [NSNumber numberWithFloat:40.0],
                          nil];
    
    [self setChartWithDataPoints:_months andValues:unitsSold];
    
    _barChartView.noDataText = @"You need to provide data for the chart.";

}

- (void) setChartWithDataPoints:(NSArray *)dataPoints andValues:(NSArray *)values {
    
    // Create an array of DataEntry objects called dataEntries
    
    NSMutableArray<ChartDataEntry *> *dataEntries = [[NSMutableArray alloc] init];
    
    for (int i = 0; i < dataPoints.count; i++) {
        NSNumber *n = [NSNumber numberWithInteger:i];
        NSNumber *value = [values objectAtIndex:n.unsignedIntegerValue];
        
        BarChartDataEntry *entry = [[BarChartDataEntry alloc] initWithValue:value.doubleValue xIndex:n.unsignedIntegerValue];
        
        [dataEntries addObject:entry];
        
        NSLog(@"%lu", dataEntries.count);
    }
    
    // Create DataSet and add dataEntries
    
    BarChartDataSet *dataSet = [[BarChartDataSet alloc] initWithYVals:dataEntries label:@"Units Sold"];
    
    // Create a ChartData object with dataSet and add to barChartView
    
    BarChartData *data = [[BarChartData alloc] initWithXVals:dataPoints dataSet:dataSet];
    
    _barChartView.data = data;
    
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: Learning Swift and Cocoa/On Off 2/On Off 2/ViewController.swift
//
//  ViewController.swift
//  On Off 2
//
//  Created by Wes Van der Klomp on 4/15/15.
//  Copyright (c) 2015 Wes Van der Klomp. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    @IBOutlet var bg: UIView!
    @IBOutlet weak var button: UIButton!
    @IBAction func button(sender: AnyObject) {
        if bg.backgroundColor == UIColor.whiteColor() {
            bg.backgroundColor = UIColor.blackColor()
            button.setTitle("On", forState: UIControlState.Normal)
        } else {
            bg.backgroundColor = UIColor.whiteColor()
            button.setTitle("Off", forState: UIControlState.Normal)
        }
    }
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        bg.backgroundColor = UIColor.whiteColor()
        button.setTitle("Off", forState: UIControlState.Normal)
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

